log4j-core/src/main/java/org/apache/logging/log4j/core/lookup/JmxRuntimeInputArgumentsLookup.java

```diff
@@ -31,17 +34,16 @@
 @Plugin(name = "jvmrunargs", category = StrLookup.CATEGORY)
 public class JmxRuntimeInputArgumentsLookup extends MapLookup {
 
-    static {
-        final List<String> argsList = ManagementFactory.getRuntimeMXBean().getInputArguments();
-        JMX_SINGLETON = new JmxRuntimeInputArgumentsLookup(MapLookup.toMap(argsList));
-    }
+    private static final Logger LOGGER = StatusLogger.getLogger();
 
-    public static final JmxRuntimeInputArgumentsLookup JMX_SINGLETON;
+    public static final JmxRuntimeInputArgumentsLookup JMX_SINGLETON = new JmxRuntimeInputArgumentsLookup();
 
     /**
      * Constructor when used directly as a plugin.
      */
-    public JmxRuntimeInputArgumentsLookup() {}
+    public JmxRuntimeInputArgumentsLookup() {
```

Comment:
*Nit:* Can't we instead have a `@PluginFactory` returning the singleton instance?

Reply:
This is not possible, since lookups are instantiated through a public no-arg constructor:

https://github.com/apache/logging-log4j2/blob/ab0fdbc51742646144f6ad29715f9bd1d9afa985/log4j-core/src/main/java/org/apache/logging/log4j/core/lookup/Interpolator.java#L84-L95